### PR TITLE
fix(2999): add validation on http query

### LINKED ISSUE #2999: feat: raise validation error when object is passed in query.
```graphql
type Query {
  findEmployees(criteria: Nested): [Employee!]!
    @http(
      baseURL: "http://localhost:8081"
      path: "/family/employees"
      query: [
        { key: "nested", value: "{{.args.criteria}}", skipEmpty: true }
      ]
    )
}

input Nested {
  maritalStatus: MaritalStatus
  hasChildren: Boolean
}
```

`nested` is an object it shouldn't be allowed inside query param, throw an validation error like `

**Expected Error**

```
Invalid query parameter type. Expected a value to be a Scalar but received an Object.
```

technical requirements:
- add unit and integration tests

### PR DESCRIPTION
**Pull request by dekkku**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #2999
/claim #2999

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/blueprint/operators/apollo_federation.rs [hunk 1]
```diff
@@ -36,8 +36,9 @@ pub fn compile_entity_resolver(inputs: CompileEntityResolver<'_>) -> Valid<IR, S
                 Resolver::Http(http) => compile_http(
                     config_module,
                     http,
+                    field,
                     // inner resolver should resolve only single instance of type, not a list
```
Comment (laststylebender14) [inline src/core/blueprint/operators/apollo_federation.rs:40]: can remove this comment if not required.

**Comment by tusharmath** [comment]
@dekkku Can we move all the mustache template validation to a separate module? This module will maintain a table of all the mustache templates and ensure its usage is correct.

**Comment by tusharmath** [comment]
Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

File: src/core/blueprint/template_validation/mod.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    config: &Config,
+    mut path_iter: impl Iterator<Item = &'a String>,
+    type_of: &str,
+    leaf_validator: impl Fn(&str) -> bool,
+) -> Valid<(), BlueprintError> {
+    match config.find_type(type_of) {
+        Some(type_def) => match path_iter.next() {
+            Some(field) => match type_def.fields.get(field) {
+                Some(field_type) => {
+                    path_validator(config, path_iter, field_type.type_of.name(), leaf_validator)
```
Comment (tusharmath) [inline src/core/blueprint/template_validation/mod.rs:21]: Potential for a stack overflow.

File: src/core/blueprint/template_validation/mod.rs [hunk 3]
```diff
@@ -57,3 +57,46 @@ pub fn validate_argument(
...
+                query: Query
+            }
+            type Query {
+                posts(id: PostData): Int @http(url: "upstream.com", query: [{key: "id", value: "{{.args.id.data}}"}])
+            }
+            type PostData {
+                author: String
+                data: PostData
+            }
+        "#;
```
Comment (tusharmath) [inline src/core/blueprint/template_validation/mod.rs:83]: use `include_config!` macro

**Review by tusharmath — APPROVED** [review]

**Review by tusharmath — APPROVED** [review]

**Pull request merged**